pkg/apis/operator/v1alpha1/types.go

```diff
@@ -87,6 +87,10 @@ type Settings struct {
 	// cluster.
 	// +optional
 	VerticalPodAutoscaler *SettingVerticalPodAutoscaler `json:"verticalPodAutoscaler,omitempty"`
+	// TopologyAwareRouting controls certain settings for topology-aware traffic routing in the cluster.
+	// See https://github.com/gardener/gardener/blob/master/docs/usage/topology_aware_routing.md.
+	// +optional
+	TopologyAwareRouting *SettingTopologyAwareRouting `json:"topologyAwareRouting,omitempty"`
```

Comment:
Does it make sense to reuse the type from the core API?

Reply:
The type for VPA is not reused - ref https://github.com/gardener/gardener/blob/d21b3ccfa1b6d76e483ba83b6f292d232ca35e55/pkg/apis/operator/v1alpha1/types.go#L100-L110.
That's why I preferred to do not reuse.

Reply:
If we reused the type also for the VPA field, would you then prefer it? 😉 If not, why not?

Reply:
In the the operator VPA setting enabled is `*bool` https://github.com/gardener/gardener/blob/d21b3ccfa1b6d76e483ba83b6f292d232ca35e55/pkg/apis/operator/v1alpha1/types.go#L109 while in the Seed VPA setting - it is `bool`
https://github.com/gardener/gardener/blob/e48872933e3039438d971786798ded799181e202/pkg/apis/core/v1beta1/types_seed.go#L311.

The go types for the Seed settings are prefixed with `SeedSetting` - `SeedSettingVerticalPodAutoscaler` and `SeedSettingTopologyAwareRouting`.
The docs strings of these fields contain Seed details. For example `SeedSettingTopologyAwareRouting` enumerates the topology-aware Services in the Seed https://github.com/gardener/gardener/blob/e48872933e3039438d971786798ded799181e202/pkg/apis/core/v1beta1/types_seed.go#L377. `SeedSettingVerticalPodAutoscaler` also contains the seed word several times https://github.com/gardener/gardener/blob/e48872933e3039438d971786798ded799181e202/pkg/apis/core/v1beta1/types_seed.go#L308-L310.

If we reuse the Seed setting types, then we should reuse for both of the settings.

Reply:
The `Enabled` field in the operator API for VPA is `*bool` because we'd like to default it to `true`, while it's `bool` in the core API because we'd like to default it to ` false` (even though we could achieve this also with `*bool`, of course).
I'm fine keeping it as is for now to avoid the additional effort with relatively little value.